### ISSUE TITLE
Segmentation fault crash report

### ISSUE THREAD

**Issue by n6g7** [pi-hole/FTL#1503]
### Versions
- Pi-hole version is v5.14.1 (Latest: v5.14.2)
- AdminLTE version is v5.17 (Latest: v5.17)
- FTL version is v5.19.1 (Latest: v5.19.2)

### Platform
- OS and version:  Raspbian GNU/Linux 11 (bullseye)
- Platform:  Raspberry Pi 2b

### Expected behavior

FTL doesn't crash.

### Actual behavior / bug

FTL crashed with a segmentation fault signal.
`tail` of `pihole-FTL.log` after the crash:
```log
[2022-12-11 08:19:48.009 22076M] Resizing "FTL-dns-cache" from 196608 to (12544 * 16) == 200704 (/dev/shm: 14.9MB used, 485.5MB total, FTL uses 14.9MB)
[2022-12-11 08:19:50.455 22076M] Resizing "FTL-strings" from 573440 to (614400 * 1) == 614400 (/dev/shm: 14.9MB used, 485.5MB total, FTL uses 14.9MB)
[2022-12-11 08:19:50.679 22076M] Resizing "FTL-dns-cache" from 200704 to (12800 * 16) == 204800 (/dev/shm: 14.9MB used, 485.5MB total, FTL uses 14.9MB)
[2022-12-11 08:21:53.343 22076M] Resizing "FTL-dns-cache" from 204800 to (13056 * 16) == 208896 (/dev/shm: 14.9MB used, 485.5MB total, FTL uses 14.9MB)
[2022-12-11 08:21:57.237 22076M] Resizing "FTL-domains" from 696320 to (35840 * 20) == 716800 (/dev/shm: 14.9MB used, 485.5MB total, FTL uses 14.9MB)
[2022-12-11 08:30:52.649 22076M] Resizing "FTL-dns-cache" from 208896 to (13312 * 16) == 212992 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 14.9MB)
[2022-12-11 08:41:58.294 22076M] Resizing "FTL-dns-cache" from 212992 to (13568 * 16) == 217088 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 14.9MB)
[2022-12-11 08:45:40.332 22076M] Resizing "FTL-dns-cache" from 217088 to (13824 * 16) == 221184 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 08:54:49.607 22076M] Resizing "FTL-dns-cache" from 221184 to (14080 * 16) == 225280 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:07:33.333 22076M] Resizing "FTL-dns-cache" from 225280 to (14336 * 16) == 229376 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:21:33.594 22076M] Resizing "FTL-domains" from 716800 to (36864 * 20) == 737280 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:22:23.507 22076M] Resizing "FTL-dns-cache" from 229376 to (14592 * 16) == 233472 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:28:00.472 22076M] Resizing "FTL-dns-cache" from 233472 to (14848 * 16) == 237568 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:28:03.529 22076M] Resizing "FTL-dns-cache" from 237568 to (15104 * 16) == 241664 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:28:05.700 22076M] Resizing "FTL-dns-cache" from 241664 to (15360 * 16) == 245760 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:28:07.205 22076M] Resizing "FTL-dns-cache" from 245760 to (15616 * 16) == 249856 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:28:40.765 22076M] Resizing "FTL-dns-cache" from 249856 to (15872 * 16) == 253952 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:29:07.386 22076M] Resizing "FTL-dns-cache" from 253952 to (16128 * 16) == 258048 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:29:19.677 22076M] Resizing "FTL-dns-cache" from 258048 to (16384 * 16) == 262144 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:29:25.023 22076M] Resizing "FTL-dns-cache" from 262144 to (16640 * 16) == 266240 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:29:39.808 22076M] Resizing "FTL-dns-cache" from 266240 to (16896 * 16) == 270336 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:30:23.409 22076M] Resizing "FTL-dns-cache" from 270336 to (17152 * 16) == 274432 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:31:55.073 22076M] Resizing "FTL-dns-cache" from 274432 to (17408 * 16) == 278528 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:32:34.942 22076M] Resizing "FTL-dns-cache" from 278528 to (17664 * 16) == 282624 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:32:42.221 22076M] Resizing "FTL-strings" from 614400 to (655360 * 1) == 655360 (/dev/shm: 15.0MB used, 485.5MB total, FTL uses 15.0MB)
[2022-12-11 09:33:35.352 22076M] Resizing "FTL-dns-cache" from 282624 to (17920 * 16) == 286720 (/dev/shm: 15.1MB used, 485.5MB total, FTL uses 15.1MB)
[2022-12-11 09:33:47.457 22076M] Resizing "FTL-domains" from 737280 to (37888 * 20) == 757760 (/dev/shm: 15.1MB used, 485.5MB total, FTL uses 15.1MB)
[2022-12-11 09:34:03.951 22076M] Resizing "FTL-dns-cache" from 286720 to (18176 * 16) == 290816 (/dev/shm: 15.1MB used, 485.5MB total, FTL uses 15.1MB)
[2022-12-11 09:34:31.411 22076M] Resizing "FTL-dns-cache" from 290816 to (18432 * 16) == 294912 (/dev/shm: 15.1MB used, 485.5MB total, FTL uses 15.1MB)
[2022-12-11 09:35:41.723 22076M] Resizing "FTL-queries" from 12435456 to (286720 * 44) == 12615680 (/dev/shm: 15.1MB used, 485.5MB total, FTL uses 15.1MB)
[2022-12-11 09:36:34.889 22076M] Resizing "FTL-dns-cache" from 294912 to (18688 * 16) == 299008 (/dev/shm: 15.3MB used, 485.5MB total, FTL uses 15.3MB)
[2022-12-11 09:39:01.446 28671/F22076] !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[2022-12-11 09:39:01.446 28671/F22076] ---------------------------->  FTL crashed!  <----------------------------
[2022-12-11 09:39:01.446 28671/F22076] !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[2022-12-11 09:39:01.446 28671/F22076] Please report a bug at https://github.com/pi-hole/FTL/issues
[2022-12-11 09:39:01.446 28671/F22076] and include in your report already the following details:
[2022-12-11 09:39:01.446 28671/F22076] FTL has been running for 37681 seconds
[2022-12-11 09:39:01.447 28671/F22076] FTL branch: master
[2022-12-11 09:39:01.447 28671/F22076] FTL version: v5.19.1
[2022-12-11 09:39:01.447 28671/F22076] FTL commit: b48b3e1f
[2022-12-11 09:39:01.447 28671/F22076] FTL date: 2022-11-14 22:01:50 +0000
[2022-12-11 09:39:01.447 28671/F22076] FTL user: started as pihole, ended as pihole
[2022-12-11 09:39:01.448 28671/F22076] Compiled for armv7hf (compiled on CI) using arm-linux-gnueabihf-gcc (Debian 8.3.0-2) 8.3.0
[2022-12-11 09:39:01.448 28671/F22076] Process details: MID: 22076
[2022-12-11 09:39:01.448 28671/F22076]                  PID: 28671
[2022-12-11 09:39:01.448 28671/F22076]                  TID: 28671
[2022-12-11 09:39:01.449 28671/F22076]                  Name: pihole-FTL
[2022-12-11 09:39:01.449 28671/F22076] Received signal: Segmentation fault
[2022-12-11 09:39:01.450 28671/F22076]      at address: 0x76fa0008
[2022-12-11 09:39:01.450 28671/F22076]      with code:  SEGV_MAPERR (Address not mapped to object)
[2022-12-11 09:39:01.451 28671/F22076] Backtrace:
[2022-12-11 09:39:01.453 28671/F22076] B[0000]: /usr/bin/pihole-FTL(generate_backtrace+0x2d) [0x45946e]
[2022-12-11 09:39:01.971 28671/F22076] L[0000]: /__w/FTL/FTL/src/signals.c:99
[2022-12-11 09:39:01.977 28671/F22076] B[0001]: /usr/bin/pihole-FTL(+0x3b7fe) [0x4597fe]
[2022-12-11 09:39:02.043 28671/F22076] L[0001]: /__w/FTL/FTL/src/signals.c:242
[2022-12-11 09:39:02.050 28671/F22076] B[0002]: /lib/arm-linux-gnueabihf/libc.so.6(__default_rt_sa_restorer+0) [0x76d5ddb0]
[2022-12-11 09:39:02.051 28671/F22076] B[0003]: /usr/bin/pihole-FTL(_lock_shm+0x55) [0x458906]
[2022-12-11 09:39:02.117 28671/F22076] L[0003]: /__w/FTL/FTL/src/shmem.c:420 (discriminator 1)
[2022-12-11 09:39:02.123 28671/F22076] B[0004]: /usr/bin/pihole-FTL(_FTL_new_query+0x1ed) [0x44e34e]
[2022-12-11 09:39:02.195 28671/F22076] L[0004]: /__w/FTL/FTL/src/dnsmasq_interface.c:628
[2022-12-11 09:39:02.203 28671/F22076] B[0005]: /usr/bin/pihole-FTL(tcp_request+0x62b) [0x47e770]
[2022-12-11 09:39:02.292 28671/F22076] L[0005]: /__w/FTL/FTL/src/dnsmasq/forward.c:2319 (discriminator 4)
[2022-12-11 09:39:02.303 28671/F22076] B[0006]: /usr/bin/pihole-FTL(+0x57072) [0x475072]
[2022-12-11 09:39:02.403 28671/F22076] L[0006]: /__w/FTL/FTL/src/dnsmasq/dnsmasq.c:2053
[2022-12-11 09:39:02.418 28671/F22076] B[0007]: /usr/bin/pihole-FTL(main_dnsmasq+0xe5b) [0x4766fc]
[2022-12-11 09:39:02.511 28671/F22076] L[0007]: /__w/FTL/FTL/src/dnsmasq/dnsmasq.c:1278
[2022-12-11 09:39:02.521 28671/F22076] B[0008]: /usr/bin/pihole-FTL(main+0xbd) [0x444a8e]
[2022-12-11 09:39:02.581 28671/F22076] L[0008]: /__w/FTL/FTL/src/main.c:118
[2022-12-11 09:39:02.587 28671/F22076] B[0009]: /lib/arm-linux-gnueabihf/libc.so.6(__libc_start_main+0x114) [0x76d45be0]
[2022-12-11 09:39:02.588 28671/F22076] ------ Listing content of directory /dev/shm ------
[2022-12-11 09:39:02.588 28671/F22076] File Mode User:Group      Size  Filename
[2022-12-11 09:39:02.588 28671/F22076] rwxrwxrwx root:root       260  .
[2022-12-11 09:39:02.589 28671/F22076] rwxr-xr-x root:root         3K  ..
[2022-12-11 09:39:02.589 28671/F22076] rw------- pihole:pihole     4K  FTL-per-client-regex
[2022-12-11 09:39:02.589 28671/F22076] rw------- pihole:pihole   299K  FTL-dns-cache
[2022-12-11 09:39:02.590 28671/F22076] rw------- pihole:pihole    12K  FTL-overTime
[2022-12-11 09:39:02.590 28671/F22076] rw------- pihole:pihole    13M  FTL-queries
[2022-12-11 09:39:02.590 28671/F22076] rw------- pihole:pihole   618K  FTL-upstreams
[2022-12-11 09:39:02.590 28671/F22076] rw------- pihole:pihole   332K  FTL-clients
[2022-12-11 09:39:02.591 28671/F22076] rw------- pihole:pihole   758K  FTL-domains
[2022-12-11 09:39:02.591 28671/F22076] rw------- pihole:pihole   655K  FTL-strings
[2022-12-11 09:39:02.591 28671/F22076] rw------- pihole:pihole    16  FTL-settings
[2022-12-11 09:39:02.592 28671/F22076] rw------- pihole:pihole   248  FTL-counters
[2022-12-11 09:39:02.592 28671/F22076] rw------- pihole:pihole    56  FTL-lock
[2022-12-11 09:39:02.592 28671/F22076] ---------------------------------------------------
[2022-12-11 09:39:02.592 28671/F22076] Please also include some lines from above the !!!!!!!!! header.
[2022-12-11 09:39:02.592 28671/F22076] Thank you for helping us to improve our FTL engine!
[2022-12-11 09:39:02.593 28671/F22076] Asking parent pihole-FTL (PID 22076) to shut down
[2022-12-11 09:39:02.593 28671/F22076] FTL fork terminated!
[2022-12-11 09:39:02.594 22076M] Received: Real-time signal 2 (36 -> 2)
[2022-12-11 09:39:02.594 28673/F22076] Error when obtaining outer SHM lock: Owner died
[2022-12-11 09:39:02.707 28673/F22076] TCP worker already terminating!
[2022-12-11 09:39:02.709 28669/F22076] Failed to unlock inner SHM lock: Operation not permitted
[2022-12-11 09:39:02.712 22076M] Shutting down...
[2022-12-11 09:39:04.101 22076M] Finished final database update (stored 57 queries)
[2022-12-11 09:39:04.102 22076M] Waiting for threads to join
[2022-12-11 09:39:04.102 22076M] Thread database (0) is idle, terminating it.
[2022-12-11 09:39:04.103 22076M] Thread housekeeper (1) is idle, terminating it.
[2022-12-11 09:39:04.103 22076M] Thread DNS client (2) is idle, terminating it.
[2022-12-11 09:39:04.104 22076M] All threads joined
[2022-12-11 09:39:04.104 22076M] Joining API worker thread 0
[2022-12-11 09:39:04.104 22076M] Joining API worker thread 1
[2022-12-11 09:39:04.104 22076M] Joining API worker thread 2
[2022-12-11 09:39:04.105 22076M] Joining API worker thread 3
[2022-12-11 09:39:04.105 22076M] Joining API worker thread 4
[2022-12-11 09:39:04.145 22076M] ########## FTL terminated after 10h 28m 3s  (code 1)! ##########
```

### Steps to reproduce

-

## Debug Token

That was generated *after* I ran `pihole restartdns`, so might not show the actual crash state.
- URL:  https://tricorder.pi-hole.net/18RtSrH8/

### Additional context

Probably unrelated but I happen to have been working on my pi-hole the day before: I'm in the process of migrating my pi-hole install to a docker container and had successfully deployed the new install yesterday, which I left running overnight. The "old" install (the one that crashed) was still running, serving DNS queries for my network and being the only active DHCP server. At first I thought there might have been a double DHCP server clash but the new install was *not* configured, meaning that DHCP was off (I double-checked that yesterday) and no client on my network should have had any reason to query it for DNS. So, despite the coincidence, I don't think this is related to the crash.

I had not made any significant configuration change to the "old" (but still in use) pi-hole install, apart from adding two CNAME records.

Running `pihole restartdns` fixed the issue.

**Comment by DL6ER** [comment]
Sorry for the inconvenience. This is a known issue with TCP forks crashing and fixed since almost one month with FTL v5.19.2. Please update.

**Comment by n6g7** [comment]
No problem, sounds like an easy fix then. Thanks for the quick reply too, I'll close the issue then.
